bundles/org.openhab.binding.modbus.e3dc/src/main/java/org/openhab/binding/modbus/e3dc/internal/modbus/Parser.java

```diff
@@ -59,17 +58,10 @@ public Parser(DataType type) {
     }
 
     public void handle(AsyncModbusReadResult result) {
-        byte[] newArray = new byte[size];
         long startTime = System.currentTimeMillis();
         Optional<ModbusRegisterArray> opt = result.getRegisters();
         if (opt.isPresent()) {
-            ModbusRegisterArray registers = opt.get();
-            int i = 0;
-            for (ModbusRegister reg : registers) {
-                System.arraycopy(reg.getBytes(), 0, newArray, i, 2);
-                i += 2;
-            }
-            setArray(newArray);
+            setArray(opt.get().getBytes());
```

Comment:
Are these equivalent?

Reply:
Yes, that should be equivalent. Tests should further give confirmation on this.

Previously it was hard to get raw bytes "abstracted" by ModbusRegisterArray and this type of copying of bytes was needed. 

A bit of modbus terminology: register means 16 bit (big endian) word returned by the modbus device. You could consider modbus being a remotely accessible memory, memory locations represented by register addresses.

Often in these bindings the register data is converted to 16/32 bit integers and floats, that we give meaning to that raw data.